state-network.md

```diff
@@ -125,134 +83,237 @@ A node should track their own radius value and provide this value in all Ping or
-- All contract bytecode for newly created contracts
+We will give the overview of the process for the account trie. The same process should be applied
+for the storage trie as well.
 
-> TODO: Figure out language for defining which trie nodes from this proof the bridge node must initialize gossip.
+The goal of the "recursive gossip" mechanism is to reduce the burden of responsibility placed on
+bridge nodes for injecting new state data into the network while simultaniously spreading the
+responsibility for gossiping new state data across the nodes in the network.
 
-> TODO: Determine mechanism for contract code.
+At each block we construct a proof for each new or modified value in the trie, which is stored in
+the leaf nodes. The bridge node would search the DHT for nodes that are *interested* in storing
+each leaf node and gossip the proof to those nodes.
 
-The receiving DHT node will propagate the data to nearby nodes from their routing table.
+> For example, let this be the proof: `[A, B, C, D, E, F]`, then `A` is the root note and `F` is
+> the target node. This proof should be gossiped to nodes that are interested in storing node `F`.
 
-### Updating cold Leaf Proofs
+The recipients of this gossip are then responsible for gossiping the penultimate node of the proof
+(parent of the target node, `E` in our example). To do so, they would strip the last (target) node
+from the proof, search the DHT for nodes that are interested in `E` and gossip this proof to them.
 
-Anytime the state root changes for either the main account trie or a contract storage trie, every leaf proof under that root will need to be updated.  The primary gossip mechanism will ensure that leaf data that was added, modified, or removed will receive and updated proof.  However, we need a mechanism for updating the leaf proofs for "cold" data that has not been changed.
+This process repeats until it terminates at the state root, with the final round of gossip only
+containing the `[A]` which is the state root node of the trie.
 
-Each time a new block is added to the chain, the DHT nodes storing leaf proof data will need to perform a walk of the trie starting at the state root. This walk of the trie will be directed towards the slice of the trie dictated by the set of leaves that the node is storing. As the trie is walked it should be compared to the previous proof from the previous state root. This walk concludes once all of the in-range leaves can be proven with the new state root.
+### Bridge Node Responsibilities
 
+The bridge is responsible for creating and gossiping all following data and their proofs:
 
-> TODO: reverse diffs and storing only the latest proof.
+- account trie data:
+    - all of the new and modified account nodes from the state trie
+- contract storage trie data:
+    - all of the new and modified storage slots from each modified contract storage trie
+    - proof has to include the proof for the account trie that corresponds to the same contract
```

Comment:
I think this is the first location in the spec that this gets mentioned? It should probably be mentioned higher up, and perhaps also explained better how it gets formatted (it's all 1 one proof or two proofs? in case of the latter how is serialization? etc.)

Reply:
I actually think it is specified.

- at line 202 we say it should contain `StorageWitness`
- at line 128 we see that `StorageWitness` contains a proof and `StateWitness` (which contains another proof)

I would say this also answer your question about serialization, right?

I wouldn't block this PR on this, but we can have a followup on how to make it clearer/less confusing

Reply:
ok, my bad, I missed that StorageWitness object. It's all good then!